### PR TITLE
[AV-154711] fix for AKO restart issue

### PR DESCRIPTION
**Pull request by swathinsankaran**
This PR fixes a crash issue during passthrough route cleanup after adding aviinfrasetting.
Also added a couple of UT fixes caused by the ingressClass optimizations.

Tested scenarios
>k8s and Openshift
1. Passthrough ingress transition from ingressClass without infraSetting to ingressClass with infraSetting
2. Passthrough ingress transition from ingressClass with infraSetting to ingressClass without infraSetting
3. Passthrough Ingress deletion

### CODE REVIEW

**Comment by swathinsankaran** [comment]
build ako

**Comment by swathinsankaran** [comment]
build ako

**Comment by swathinsankaran** [comment]
build ako

**Review by apalsule — APPROVED** [review]

**Review by aaha97 — DISMISSED** [review]
LGTM

**Comment by swathinsankaran** [comment]
build ako

**Comment by swathinsankaran** [comment]
build ako

**Comment by swathinsankaran** [comment]
build ako

**Review by akshayhavile — COMMENTED** [review]
@swathinsankaran: We will require different type of changes to address this.

File: internal/nodes/avi_model_evh_nodes.go [hunk 1]
```diff
@@ -1832,6 +1838,9 @@ func (o *AviObjectGraph) DeletePoolForHostnameForEvh(vsName, hostname string, ro
 	namespace := routeIgrObj.GetNamespace()
 	ingName := routeIgrObj.GetName()
 	vsNode := o.GetAviEvhVS()
+	if len(vsNode) == 0 {
```
Comment (akshayhavile) [inline internal/nodes/avi_model_evh_nodes.go:1841]: I think we can remove this code now.

Comment (swathinsankaran) [reply internal/nodes/avi_model_evh_nodes.go]: I have removed the code.

**Review by akshayhavile — APPROVED** [review]

**Review by aaha97 — APPROVED** [review]
LGTM

**Pull request merged**